### PR TITLE
Refactored job preferences

### LINKED ISSUE #8235: Occupation Preferences issue
when you have a job set to "high", if you right-click any **_other**_ job the job on "high" goes to "medium".
example:
you have scientist to "medium" and roboticist to "high";
right-click scientist;
scientist goes to "low" and roboticist goes to "medium".

### PR DESCRIPTION
**Pull request by DamianX**
Another in the series of "I wanted to make a change but the existing cripplingly bad code prevented me from doing it so here's a refactor"
Accidentally fixes #8235

:cl:
* tweak: All job preferences have been reset because the way they are stored in the database has changed.
* bugfix: Right-clicking any other job in the preferences window while there was a job set to "high" no longer sets that "high" job to medium.

### CODE REVIEW

**Comment by DamianX** [comment]
oh yeah this will reset all job preferences

File: code/game/jobs/job_controller.dm [hunk 1]
```diff
@@ -280,7 +280,7 @@ var/global/datum/controller/occupations/job_master
 
 	// Loop through all levels from high to low
 	var/list/shuffledoccupations = shuffle(occupations)
-	for(var/level = 1 to 3)
+	for(var/level = 3; level > 0; level--)
```
Comment (Exxion) [inline code/game/jobs/job_controller.dm]:
```suggestion
	for(var/level = 3 to 1 step -1)
```
I don't really want to copy this to every other instance
EDIT: I thought I saw more than one other instance

**Comment by Apogee-dev** [comment]
Quick question, does this affect any of the existing icon preview bugs in character setup? I was working on a fix for those but noticed we're fucking with the same files here.

**Comment by DamianX** [comment]
It doesn't change how it works, just uses the datum types rather than the value of the `flag` var.

**Comment by Apogee-dev** [comment]
> It doesn't change how it works, just uses the datum types rather than the value of the `flag` var.

So it doesn't change anything about the preview icons? Alright, I'll hang onto what I've got and drop it in later then.

**Comment by Dacendeth** [comment]
Does this mean people with old slots like plasmaman hos and other now illegal things will be kicked out of their old slots?

**Comment by DamianX** [comment]
correct

**Comment by Kurfursten** [comment]
Please include a CL that notes player job prefs are reset

**Comment by Exxion** [comment]
Is it possible to set something up to migrate old job prefs to this new system?

**Comment by DamianX** [comment]
Yes, the data is even still in the database, but I'm not willing to write code to do that.

**Pull request merged**